### PR TITLE
Update working groups

### PR DESCRIPTION
**Pull request by 1ucian0**
Adding the following groups, as mandated by the TSC
 * Generics and circuit families
 * Grammar
 * Types and casting

### CODE REVIEW

File: WG.md [hunk 1]
```diff
@@ -8,12 +8,42 @@ for review by the TSC at an agreed upon date. A WG will be automatically disband
...
+**Questions**:
+
+ * Any implicit casts?
+ * What explicit casts are allowed?
+ * Registers vs arrays?
+ * Can you index into integers and get bit value?
+ * Is an int equivalent to an array of bits?
+
+Chair: Michael Healy (IBM Quantum)  
+Members: Niels, Hiroshi, Blake Johnson, Colm Ryan, Luciano Bello, Prasahnt Sivarajah
```
Comment (shiyunon) [inline WG.md:37]: @1ucian0 I think my name is missing here. could you add me to here as well?

Comment (1ucian0) [reply WG.md]: Done in 8784e49

File: WG.md [hunk 2]
```diff
@@ -8,12 +8,43 @@ for review by the TSC at an agreed upon date. A WG will be automatically disband
...
+### Generics and circuit families
+
+**Objective**: ...
+
+Chair: Niel de Beaudrap  
+Members: Colm Ryan, Andrew Cross, Ali Javadi, Blake Johnson
+
+### Grammar
+
+**Objective**:
```
Comment (aspcompiler) [inline WG.md:20]: I suggest that the Objective of this group to be "Improve ANTLR reference implementation of OpenQASM grammar".

File: WG.md [hunk 3]
```diff
@@ -8,12 +8,43 @@ for review by the TSC at an agreed upon date. A WG will be automatically disband
...
+
+**Objective**: ...
+
+Chair: Niel de Beaudrap  
+Members: Colm Ryan, Andrew Cross, Ali Javadi, Blake Johnson
+
+### Grammar
+
+**Objective**: 
+
```
Comment (aspcompiler) [inline WG.md:21]: Some of the **Questions** to ask in this group that comes to my mind:

* Are their pain points in the grammar?
* Are their places we can improve?
* Can we improve the performance?
* Can we improve the style?

**Review by aspcompiler — COMMENTED** [review]
@1ucian0 , per your request, I added a couple of comments on the Objective of grammar working group.

File: WG.md [hunk 4]
```diff
@@ -8,12 +8,43 @@ for review by the TSC at an agreed upon date. A WG will be automatically disband
 
 ## Current Working Groups
 
+### Generics and circuit families
+
+**Objective**: ...
+
+Chair: Niel de Beaudrap  
+Members: Colm Ryan, Andrew Cross, Ali Javadi, Blake Johnson
```
Comment (ndebeaudrap) [inline WG.md:16]: I think we should include Matthew Amy here.

File: WG.md [hunk 5]
```diff
@@ -8,12 +8,43 @@ for review by the TSC at an agreed upon date. A WG will be automatically disband
 
 ## Current Working Groups
 
+### Generics and circuit families
+
+**Objective**: ...
```
Comment (ndebeaudrap) [inline WG.md:13]: While I haven't had the opportunity to attend a TSC meeting, I don't believe we have been handed down a specific remit from above. My understanding is that our purpose is to recommend what range of generics / parameterised circuit family functionality (beyond angle parameters) would be suitable for inclusion in OpenQASM, and what syntax that should be used for it.

File: WG.md [hunk 6]
```diff
@@ -8,12 +8,43 @@ for review by the TSC at an agreed upon date. A WG will be automatically disband
...
 
-Objective: Define a pulse grammar "openpulse" to be used for microcoding of gate instructions with
+**Objective**: Define a pulse grammar "openpulse" to be used for microcoding of gate instructions with
 OpenQASM `defcal`'s.
-Chair: Thomas Alexander (IBM Quantum)
-Members: Blake Johnson, Colm Ryan, Derek Bolt, Peter Karalekas, Lauren Capelluto, Michael Healy,
-Prasahnt Sivarajah, Yunong Shi, Steven Heidel, Zachary Schoenfeld
+
+Chair: Thomas Alexander (IBM Quantum)  
+Members: Blake Johnson, Colm Ryan, Derek Bolt, Peter Karalekas, Lauren Capelluto, Michael Healy, Prasahnt Sivarajah, Yunong Shi, Steven Heidel, Zachary Schoenfeld
```
Comment (taalexander) [inline WG.md:47]:
```suggestion
Members: Blake Johnson, Colm Ryan, Derek Bolt, Peter Karalekas, Lauren Capelluto, Michael Healy, Prasahnt Sivarajah, Yunong Shi, Steven Heidel
```

**Review by taalexander — DISMISSED** [review]

File: WG.md [hunk 7]
```diff
@@ -8,12 +8,43 @@ for review by the TSC at an agreed upon date. A WG will be automatically disband
...
+
+Chair: Niel de Beaudrap  
+Members: Colm Ryan, Andrew Cross, Ali Javadi, Blake Johnson
+
+### Grammar
+
+**Objective**: 
+
+Chair: Li Chen  
+Members: Hiroshi Horii, Michael Healy, Luciano Bello, Jeff Heckey, Yunong Shi, Steve Heidel
```
Comment (stevenheidel) [inline WG.md:23]:
```suggestion
Members: Hiroshi Horii, Michael Healy, Luciano Bello, Jeff Heckey, Yunong Shi, Steven Heidel
```

File: WG.md [hunk 8]
```diff
@@ -8,12 +8,43 @@ for review by the TSC at an agreed upon date. A WG will be automatically disband
...
 
+### Generics and circuit families
+
+**Objective**: ...
+
+Chair: Niel de Beaudrap  
+Members: Colm Ryan, Andrew Cross, Ali Javadi, Blake Johnson
+
+### Grammar
+
```
Comment (stevenheidel) [inline WG.md:19]:
```suggestion

Note: this working group will begin working in September 2021.

```

Discussed with @aspcompiler that we would like to kick this off in a few weeks.

**Review by stevenheidel — APPROVED** [review]
LGTM

File: WG.md (hunk 4 above)
Comment (levbishop) [inline WG.md:16]:
```suggestion
Members: Colm Ryan, Andrew Cross, Ali Javadi, Blake Johnson, Matthew Amy
```

**Comment by stevenheidel** [comment]
I think we can merge this and continue iterating in future MRs. In particular we will clarify the scope of the grammar working group once it kicks off in September.

File: WG.md [hunk 9]
```diff
@@ -8,12 +8,45 @@ for review by the TSC at an agreed upon date. A WG will be automatically disband
...
+**Questions**:
+
+ * Any implicit casts?
+ * What explicit casts are allowed?
+ * Registers vs arrays?
+ * Can you index into integers and get bit value?
+ * Is an int equivalent to an array of bits?
+
+Chair: Michael Healy (IBM Quantum)  
+Members: Niel de Beaudrap, Hiroshi Horii, Blake Johnson, Colm Ryan, Luciano Bello, Prasahnt Sivarajah, Yunong Shi
```
Comment (blakejohnson) [inline WG.md:39]:
```suggestion
Members: Niel de Beaudrap, Hiroshi Horii, Blake Johnson, Colm Ryan, Luciano Bello, Prasahnt Sivarajah, Yunong Shi, Jake Lishman
```

**Comment by 1ucian0** [comment]
Hi @aspcompiler ,
   TSC resolved to merge this PR as it is and formally define the objective of the grammar WG in a following PR. Do you mind creating such a PR with your suggestion?

**Comment by aspcompiler** [comment]
> Hi @aspcompiler ,
> TSC resolved to merge this PR as it is and formally define the objective of the grammar WG in a following PR. Do you mind creating such a PR with your suggestion?

Will do. Thank you, @1ucian0 !

**Review by jakelishman — APPROVED** [review]

**Review by blakejohnson — APPROVED** [review]

File: WG.md (hunk 2 above)
Comment (1ucian0) [reply WG.md:22]: https://github.com/Qiskit/openqasm/pull/271

**Pull request merged**